include/sas.h

```diff
@@ -397,8 +469,22 @@ class SAS
 
   static sas_log_callback_t* _log_callback;
 
-private:
+  /// Converts the format of a log raised within the SAS-Client to that expected by
+  /// the common log callback, and calls the common log callback with the converted
+  /// arguments.
+  static void sasclient_log_callback(log_level_t level,
+                                     const char *module,
+                                     int line_number,
+                                     const char *fmt,
+                                     ...);
+
+  static void _sasclient_log_callback(log_level_t level,
+                                      const char *module,
+                                      int line_number,
+                                      const char *fmt,
+                                      va_list args);
```

Comment:
Why is this needed? It's the job of `sasclient_log_callback` to convert between it's  parameters and the parameters that the callback needs so I don;t see why we need to go via this function.